reactioncommerce/reaction
# Fix nnnnat remove products on grid items

Resolves #3938
Impact: **major**  
Type: **performance**

## Issue
The ProductGrid component was using a spread operator to apply all of it's own props to each ProductGridItems, this was causing the products array and several other unneeded sets of data to be added to each component's props.

## Solution
* Replaced the spread operator with explicit props for the ProductGridItems component
* Updated the key to use the `product._id` instead of the `.map` index. [ref](https://reactjs.org/docs/lists-and-keys.html#keys)
* Refactored the `renderProductGridItems` to not be an arrow function and to not accept arguments since the products are available on the component's props.
* Updated the ProductGrid Proptypes.
* Fixed a small eslint issue in an unrelated file.

## Testing
*You'll need [React devtools](https://github.com/facebook/react-devtools) installed to reproduce*
 1. Spin up a fresh RC shop.
 2. Open React devtools and inspect the ProductGridItems component. (The node will be called Reaction(ProductGridItems))
 3. Verify that the products, productIds, and productsByKey are no longer in the component props.
